src/traces/contour/hover.js

```diff
@@ -9,8 +9,26 @@
 
 'use strict';
 
+var Color = require('../../components/color');
+
 var heatmapHoverPoints = require('../heatmap/hover');
 
 module.exports = function hoverPoints(pointData, xval, yval, hovermode, hoverLayer) {
-    return heatmapHoverPoints(pointData, xval, yval, hovermode, hoverLayer, true);
+    var hoverData = heatmapHoverPoints(pointData, xval, yval, hovermode, hoverLayer, true);
+
+    if(hoverData) {
```

Comment:
Looks like `contourcarpet` does not have a `hoverPoints` handler. Could it now reuse this one here?

Reply:
I'm guessing it will take some work to sort out hover for `contourcarpet`. Unlike `scattercarpet` where you just have to look at the final position of each point, `contourcarpet` we would actually have to invert (x,y) back to (a,b), which I don't think has been implemented (it's in principle multi-valued anyway).